### PR TITLE
Initial commit for review

### PR DESCRIPTION
**Pull request by kdowney-cloudwallcapital**
SERENITY-2129

### CODE REVIEW

**Review by kmgivens — APPROVED** [review]
It looks as we discussed on the call.  I think at the next iteration we will need more results in the OptionValuationResult object to explain the definitions of the greeks, delta in particular.

File: src/python/serenity_types/pricing/derivatives/options/volsurface.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    """
+    Representation of the volatility surface, e.g. raw inputs vs. fitted surface.
+    """
+
+    RAW = "RAW"
+    """
+    A volatility surface containing the raw inputs: the option prices, IV's, etc..
+    """
+
+    FITTED = "FITTED"
```
Comment (indra-cloudwall) [inline src/python/serenity_types/pricing/derivatives/options/volsurface.py:37]: I guess model need to be updated from INTERPOLATED to FITTED then?

File: src/python/serenity_types/pricing/derivatives/options/volsurface.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    """
+    Relative value of strike vs. current spot, with log transformation, e.g. 1.05
+    """
+
+
+class VolatilitySurface(BaseModel):
+    """
+    Base type for both RAW and INTERPOLATED yield curve representations: a term structure.
+    """
+
```
Comment (indra-cloudwall) [inline src/python/serenity_types/pricing/derivatives/options/volsurface.py:63]: Based on our email discussion, should we add UUID here?

File: src/python/serenity_types/pricing/derivatives/rates/yield_curve.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    """
+    The input DF value, if rate not provided.
+    """
+
+
+class YieldCurve(BaseModel):
+    """
+    Base type for both RAW and INTERPOLATED yield curve representations: a term structure.
+    """
+
```
Comment (indra-cloudwall) [inline src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:133]: Based on our email discussion, should we add UUID here?

File: src/python/serenity_types/pricing/derivatives/options/volsurface.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    Absolute value of strike, e.g. the 20000 option.
+    """
+
+    LOG_MONEYNESS = "LOG_MONEYNESS"
+    """
+    Relative value of strike vs. current spot, with log transformation, e.g. 1.05
+    """
+
+
+class VolatilitySurface(BaseModel):
```
Comment (indra-cloudwall) [inline src/python/serenity_types/pricing/derivatives/options/volsurface.py:59]: Is this the expected return for getVolatilitySurfaces? or we need another VolSurfaceDefinition?
```

class VolSurfaceDefinition(BaseModel):
    """
    Basic metadata for volatility surfaces
    """

    vol_surface_id: UUID
    """
    The unique identifier for this surface
    """

    display_name: str
    """
    Human-readable descrition of this curve, e.g. OIS (RAW) or OIS (Interpolated, FLAT_FWD).
    """

    vol_model: VolModel
    """
    Volatility model used for this surface.
    """

    strike_type: StrikeType
    """
    Strike representation used for this surface, e.g. ABSOLUTE or LOG_MONEYNESS.
    """

    start_datetime: datetime
    """
    UTC datetime for the earliest version in the database.
    """

    end_datetime: datetime
    """
    UTC datetime for the latest version in the database.
    """

```

File: src/python/serenity_types/pricing/derivatives/rates/yield_curve.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    The input rate value, if DF not provided.
+    """
+
+    discount_factor: float
+    """
+    The input DF value, if rate not provided.
+    """
+
+
+class YieldCurve(BaseModel):
```
Comment (indra-cloudwall) [inline src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:129]: Same as above, do we need YieldCurveDefinition?

```
class YieldCurveDefinition(BaseModel):
    """
    Basic metadata for yield curfe.
    """

    yield_curve_id: UUID
    """
    The unique identifier for this surface
    """

    display_name: str
    """
    Human-readable descrition of this curve, e.g. OIS (RAW) or OIS (Interpolated, FLAT_FWD)
    """

    start_datetime: datetime
    """
    UTC datetime for the earliest version in the database.
    """

    end_datetime: datetime
    """
    UTC datetime for the latest version in the database.
    """

```

File: src/python/serenity_types/pricing/derivatives/options/volsurface.py (hunk 1 above)
Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/options/volsurface.py:37]: What do you mean? "model" here?

File: src/python/serenity_types/pricing/derivatives/options/volsurface.py (hunk 2 above)
Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/options/volsurface.py:63]: Yes, I will add today.

File: src/python/serenity_types/pricing/derivatives/rates/yield_curve.py (hunk 3 above)
Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:133]: Yes, I will add today.

File: src/python/serenity_types/pricing/derivatives/options/volsurface.py (hunk 4 above)
Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/options/volsurface.py:59]: You are right, I need to revisit the return values for the getVolatilitySurfaces and getYieldCurves calls. I will do this as part of the work today to incorporate the new types into the serenity.specifications notebook. Bear with me.

File: src/python/serenity_types/pricing/derivatives/rates/yield_curve.py (hunk 5 above)
Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:129]: Yes, will do this one as well.

File: src/python/serenity_types/pricing/derivatives/options/volsurface.py (hunk 1 above)
Comment (indra-cloudwall) [reply src/python/serenity_types/pricing/derivatives/options/volsurface.py:37]: Current Kevin's code is expecting VolSurfaceType "RAW" or "INTERPOLATED"

Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/options/volsurface.py:37]: I will make the specification consistent and use INTERPOLATED, same as @kmgivens.

File: src/python/serenity_types/pricing/derivatives/rates/yield_curve.py (hunk 5 above)
Comment (kdowney-cloudwallcapital) [reply src/python/serenity_types/pricing/derivatives/rates/yield_curve.py:129]: Same as for VolatilitySurface, we can reuse YieldCurve here, no need for separate definition class.

**Pull request merged**